diffkemp/simpll/DifferentialFunctionComparator.cpp

```diff
@@ -693,6 +699,46 @@ int DifferentialFunctionComparator::cmpGlobalValues(GlobalValue *L,
         return L != R;
 }
 
+/// Specific comparing of structure field access.
+int DifferentialFunctionComparator::cmpFieldAccess(const Function *L,
+                                                   const Function *R) const {
+    // First compute the complete offset of all GEPs in both functions.
+    // It is possible that although the process contains unpacking several
+    // anonymous unions and structs, the offset stays the same, which means that
+    // without a doubt this is semantically equal (and probably would, in fact,
+    // be also equal in machine code).
+    uint64_t OffsetL = 0, OffsetR = 0;
+    for (auto &Inst : *(L->begin())) {
```

Comment:
This loop is exactly the same as the following one. Can we make it a function (something like `accumulateAllOffsets`)?